src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpLogExporterHelperExtensions.cs

```diff
@@ -106,29 +109,280 @@ public static OpenTelemetryLoggerOptions AddOtlpExporter(this OpenTelemetryLogge
+    /// <param name="builder"><see cref="LoggerProviderBuilder"/> builder to use.</param>
+    /// <param name="configureExporterAndProcessor">Callback action for
+    /// configuring <see cref="OtlpExporterOptions"/> and <see
+    /// cref="LogRecordExportProcessorOptions"/>.</param>
+    /// <returns>The instance of <see cref="LoggerProviderBuilder"/> to chain the calls.</returns>
+#if NET8_0_OR_GREATER
+    [Experimental(DiagnosticDefinitions.LoggerProviderExperimentalApi, UrlFormat = DiagnosticDefinitions.ExperimentalApiUrlFormat)]
+#endif
+    public
+#else
+    internal
+#endif
+        static LoggerProviderBuilder AddOtlpExporter(this LoggerProviderBuilder builder, Action<OtlpExporterOptions, LogRecordExportProcessorOptions> configureExporterAndProcessor)
+        => AddOtlpExporter(builder, name: null, configureExporterAndProcessor);
+
+#if EXPOSE_EXPERIMENTAL_FEATURES
+    /// <summary>
+    /// Adds OpenTelemetry Protocol (OTLP) exporter to the LoggerProvider.
+    /// </summary>
+    /// <remarks><b>WARNING</b>: This is an experimental API which might change or be removed in the future. Use at your own risk.</remarks>
+    /// <param name="builder"><see cref="LoggerProviderBuilder"/> builder to use.</param>
+    /// <param name="name">Optional name which is used when retrieving options.</param>
+    /// <param name="configureExporter">Optional callback action for configuring <see cref="OtlpExporterOptions"/>.</param>
+    /// <returns>The instance of <see cref="LoggerProviderBuilder"/> to chain the calls.</returns>
+#if NET8_0_OR_GREATER
+    [Experimental(DiagnosticDefinitions.LoggerProviderExperimentalApi, UrlFormat = DiagnosticDefinitions.ExperimentalApiUrlFormat)]
+#endif
+    public
+#else
+    /// <summary>
+    /// Adds OpenTelemetry Protocol (OTLP) exporter to the LoggerProvider.
+    /// </summary>
+    /// <param name="builder"><see cref="LoggerProviderBuilder"/> builder to use.</param>
+    /// <param name="name">Optional name which is used when retrieving options.</param>
+    /// <param name="configureExporter">Optional callback action for configuring <see cref="OtlpExporterOptions"/>.</param>
+    /// <returns>The instance of <see cref="LoggerProviderBuilder"/> to chain the calls.</returns>
+    internal
+#endif
+        static LoggerProviderBuilder AddOtlpExporter(
+        this LoggerProviderBuilder builder,
+        string? name,
+        Action<OtlpExporterOptions>? configureExporter)
+    {
+        var finalOptionsName = name ?? Options.DefaultName;
+
+        builder.ConfigureServices(services =>
+        {
+            if (name != null && configureExporter != null)
+            {
+                // If we are using named options we register the
+                // configuration delegate into options pipeline.
+                services.Configure(finalOptionsName, configureExporter);
```

Comment:
For the other AddOtlpExporter method which takes in an `Action<OtlpExporterOptions, LogRecordExportProcessorOptions>`, we call the configure action on the options in the `LoggerProviderBuilder.AddProcessor` method regardless of whether the name is null or not.

Could we be consistent with that? Is there any benefit to configuring options differently in this method?

Reply:
This is why I don't like the pattern where we expose a delegate which takes two parameters (ex `Action<OtlpExporterOptions, LogRecordExportProcessorOptions>`). There's no way to do that correctly with Options API. You can only execute it inline which creates some rough edges for users who expect "Options API things" to work. For example there is the `IPostConfigureOptions` thing where you can register some delegate to fire after other configurations. The inline call won't play nice with that. It will always be after all the configure calls and the post configure calls. Essentially it is inconsistent because we try to support Options API fully where we can.

Reply:
Thanks for the explanation!